scripts/publish_orb.sh

```diff
@@ -1,60 +1,142 @@
+# used to publish a dev version of the orb. That can be pulled in using
+# artsy/<orb-name>@dev:<version>. This is useful for testing purposes.
+#
+# This will be referred to as "dev mode" in later comments
+DEV=""
+if [ "$BRANCH" != "master" ]; then
+  DEV="dev:"
+  echo $(YELLOW "[Running in dev mode]")
+fi
+
+# When in dev mode
+if [ ! -z "$DEV" ]; then
+  echo ""
+  echo "This will be a dev deployment (prefixed with dev:)"
 fi
 
-echo ""
-echo "Beginning publish of artsy/$1 orb"
 
 ORB="$1"
 
-# Ensuring orb is valid
+# Ensure the orb is valid
 ./scripts/validate_orb.sh $ORB
 
 ORB_PATH=$(get_orb_path $ORB)
 VERSION=$(get_orb_version $ORB)
 IS_PUBLISHED=$(is_orb_published $ORB)
 
+# If the orb has been previously published (i.e. it already exists in cicle's registry)
 if [ ! -z "$IS_PUBLISHED" ]; then
 
   LAST_PUBLISHED=$(get_published_orb_version $ORB)
 
   case $(compare_version $VERSION $LAST_PUBLISHED) in
     "=")
-      echo "artsy/$ORB@$VERSION is the latest, skipping publish"
-      exit 0
+      # When not in dev mode
+      if [ -z "$DEV" ]; then
+        echo "artsy/$ORB@$VERSION is the latest, skipping publish"
+        exit 0
+      fi
       ;;
     "<")
-      echo "artsy/$ORB@$LAST_PUBLISHED is the latest, cannot publish older version $VERSION"
-      echo "Please update $ORB_PATH to have a version greater than $LAST_PUBLISHED"
+      echo $(RED "artsy/$ORB@$LAST_PUBLISHED is the latest, cannot publish older version $VERSION")
+      echo $(RED "Please update $ORB_PATH to have a version greater than $LAST_PUBLISHED")
       exit 1
       ;;
     ">")
-      echo "Preparing to bump artsy/$ORB from $LAST_PUBLISHED to $VERSION"
+      # when not in dev mode
+      if [ -z "$DEV" ]; then
+        echo "Preparing to bump artsy/$ORB from $LAST_PUBLISHED to $VERSION"
+      fi
       ;;
     *)
-      echo "Version comparison for artsy/$ORB failed."
-      echo "Current version: $VERSION"
-      echo "Published version: $LAST_PUBLISHED"
+      echo $(RED "Version comparison for artsy/$ORB failed.")
+      echo $(RED "Current version: $VERSION")
+      echo $(RED "Published version: $LAST_PUBLISHED")
       exit 1
       ;;
   esac
 
+  # When in dev mode
+  if [ ! -z "$DEV" ];then
+    echo "Preparing to publish dev orb artsy/$ORB@$DEV$VERSION"
+  fi
+
 else
   echo "Orb artsy/$ORB isn't in the registry. Creating its registry entry..."
   circleci orb create artsy/$ORB $TOKEN --no-prompt
+  echo "Orb created, prepaing to publish artsy/$ORB@$DEV$VERSION"
 fi
 
-circleci orb publish $ORB_PATH artsy/$ORB@$VERSION $TOKEN
 
-./slack \
-  -color "good" \
-  -title "Circle CI $ORB orb v$VERSION published!" \
-  -title_link "${CIRCLE_BUILD_URL:-https://circleci.com/gh/artsy/orbs/tree/master}" \
-  -user_name "artsyit" \
-  -icon_emoji ":crystal_ball:"
+# Publish to CircleCI (when it's not a dry run)
+if [ -z "$DRY_RUN" ]; then
+  circleci orb publish $ORB_PATH artsy/$ORB@$DEV$VERSION $TOKEN
```

Comment:
As discussed, adding a PR # to the orb tag would ensure multiple open PRs wouldn't overwrite one another